### PR TITLE
Make sure video is running during memory balloon

### PR DESCRIPTION
**Pull request by suqinhuang**
Signed-off-by: Suqin Huang <shuang@redhat.com>
id: 1130458

### CODE REVIEW

**Comment by pingl2016** [comment]
Found the below error when test the case with win10 and win2008.
07:20:39 ERROR| Reproduced traceback from: /usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/avocado_vt/test.py:420
07:20:39 ERROR| Traceback (most recent call last):
07:20:39 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/avocado_vt/test.py", line 196, in runTest
07:20:39 ERROR|     self._runTest()
07:20:39 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/avocado_vt/test.py", line 337, in _runTest
07:20:39 ERROR|     run_func(self, params, env)
07:20:39 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/virttest/error_context.py", line 135, in new_fn
07:20:39 ERROR|     return fn(*args, **kwargs)
07:20:39 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_stress.py", line 86, in run
07:20:39 ERROR|     min_sz, max_sz = balloon_test.get_memory_boundary()
07:20:39 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_check.py", line 158, in get_memory_boundary
07:20:39 ERROR|     self.ori_gmem = self.get_memory_status()
07:20:39 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_check.py", line 305, in get_memory_status
07:20:39 ERROR|     return int(self.get_used_mem())
07:20:39 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_check.py", line 243, in get_used_mem
07:20:39 ERROR|     return self.get_guest_used_mem(self.vm)
07:20:39 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/virttest/utils_test/qemu/__init__.py", line 480, in get_guest_used_mem
07:20:39 ERROR|     return utils_misc.get_used_mem(session, os_type)
07:20:39 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/virttest/utils_misc.py", line 2664, in get_used_mem
07:20:39 ERROR|     output = session.cmd_output(cmd)
07:20:39 ERROR|   File "/usr/lib/python2.6/site-packages/aexpect/client.py", line 1057, in cmd_output
07:20:39 ERROR|     raise ShellTimeoutError(cmd, o)
07:20:39 ERROR| ShellTimeoutError: Timeout expired while waiting for shell command to complete: 'systeminfo'    (output: '')

**Comment by pingl2016** [comment]
Found below issue when test the case with win8.1 and win2012r2 on rhel6 host
08:13:20 INFO | Context: Change VM memory to 4756
08:13:20 DEBUG| (monitor qmpmonitor1) Sending command 'balloon'
08:13:20 DEBUG| Send command: {'execute': 'balloon', 'arguments': {'value': 4987027456}, 'id': 'AFUYDiMi'}
08:13:21 INFO | BALLOON_CHANGE QMP event received
08:13:21 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:21 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'K64iVf8u'}
08:13:21 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:21 DEBUG| (monitor qmpmonitor1)    actual: 15665725440
08:13:22 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:22 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'QmG7RcWQ'}
08:13:22 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:22 DEBUG| (monitor qmpmonitor1)    actual: 15500050432
08:13:23 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:23 DEBUG| Send command: {'execute': 'query-balloon', 'id': '6Kxx3n9R'}
08:13:23 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:23 DEBUG| (monitor qmpmonitor1)    actual: 15388901376
08:13:24 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:24 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'fl1gLWLj'}
08:13:24 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:24 DEBUG| (monitor qmpmonitor1)    actual: 15286140928
08:13:25 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:25 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'U4zAZcUE'}
08:13:25 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:25 DEBUG| (monitor qmpmonitor1)    actual: 15183380480
08:13:26 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:26 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'cefUz6pl'}
08:13:26 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:26 DEBUG| (monitor qmpmonitor1)    actual: 15084814336
08:13:27 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:27 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'oXGyMywF'}
08:13:27 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:27 DEBUG| (monitor qmpmonitor1)    actual: 14994636800
08:13:28 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:28 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'u0YCShOf'}
08:13:28 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:28 DEBUG| (monitor qmpmonitor1)    actual: 14914945024
08:13:29 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:29 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'Vf8j1Fs8'}
08:13:29 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:29 DEBUG| (monitor qmpmonitor1)    actual: 14824767488
08:13:30 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:30 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'r2L1kJ5C'}
08:13:30 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:30 DEBUG| (monitor qmpmonitor1)    actual: 14742978560
08:13:31 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:31 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'kGT2vRrU'}
08:13:31 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:31 DEBUG| (monitor qmpmonitor1)    actual: 14652801024
08:13:32 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:32 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'NYMkfxcT'}
08:13:32 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:32 DEBUG| (monitor qmpmonitor1)    actual: 14497611776
08:13:33 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:33 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'UJ3sACPZ'}
08:13:33 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:33 DEBUG| (monitor qmpmonitor1)    actual: 14327742464
08:13:34 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:34 DEBUG| Send command: {'execute': 'query-balloon', 'id': '0nJn6nG7'}
08:13:34 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:34 DEBUG| (monitor qmpmonitor1)    actual: 14212399104
08:13:35 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:35 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'iKt2u7Va'}
08:13:35 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:35 DEBUG| (monitor qmpmonitor1)    actual: 14057209856
08:13:36 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:36 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'I1gC6fnC'}
08:13:36 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:36 DEBUG| (monitor qmpmonitor1)    actual: 13885243392
08:13:37 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:37 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'PkTQSBs7'}
08:13:37 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:37 DEBUG| (monitor qmpmonitor1)    actual: 13686013952
08:13:38 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:38 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'oyugZ30D'}
08:13:38 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:38 DEBUG| (monitor qmpmonitor1)    actual: 13522436096
08:13:39 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:39 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'J4A2ypgZ'}
08:13:39 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:39 DEBUG| (monitor qmpmonitor1)    actual: 13400801280
08:13:40 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:40 DEBUG| Send command: {'execute': 'query-balloon', 'id': '2FljAPXN'}
08:13:40 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:40 DEBUG| (monitor qmpmonitor1)    actual: 13283360768
08:13:41 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:41 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'uOvdhavg'}
08:13:41 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:41 DEBUG| (monitor qmpmonitor1)    actual: 13163823104
08:13:42 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:42 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'uxJUxTTf'}
08:13:42 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:42 DEBUG| (monitor qmpmonitor1)    actual: 13054771200
08:13:43 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:43 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'NBEUhx1y'}
08:13:43 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:43 DEBUG| (monitor qmpmonitor1)    actual: 12960399360
08:13:44 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:44 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'JRDt7UzR'}
08:13:44 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:44 DEBUG| (monitor qmpmonitor1)    actual: 12907970560
08:13:45 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:45 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'ciACTWvf'}
08:13:45 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:45 DEBUG| (monitor qmpmonitor1)    actual: 12815695872
08:13:46 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:46 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'OHpKH7EI'}
08:13:46 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:46 DEBUG| (monitor qmpmonitor1)    actual: 12719226880
08:13:47 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:47 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'BNxeEYyG'}
08:13:47 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:47 DEBUG| (monitor qmpmonitor1)    actual: 12618563584
08:13:48 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:48 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'oNGJqElT'}
08:13:48 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:48 DEBUG| (monitor qmpmonitor1)    actual: 12528386048
08:13:49 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:49 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'zgiVVBd4'}
08:13:49 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:49 DEBUG| (monitor qmpmonitor1)    actual: 12429819904
08:13:50 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:50 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'nENj8VD4'}
08:13:50 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:50 DEBUG| (monitor qmpmonitor1)    actual: 12320768000
08:13:51 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:51 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'Kn2cfTQk'}
08:13:51 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:51 DEBUG| (monitor qmpmonitor1)    actual: 12150898688
08:13:52 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:52 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'PN1nvJ9f'}
08:13:52 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:52 DEBUG| (monitor qmpmonitor1)    actual: 12025069568
08:13:53 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:53 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'rIj8aqOx'}
08:13:53 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:53 DEBUG| (monitor qmpmonitor1)    actual: 11924406272
08:13:54 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:54 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'FaZNghUH'}
08:13:54 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:54 DEBUG| (monitor qmpmonitor1)    actual: 11834228736
08:13:55 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:55 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'M21enPSN'}
08:13:55 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:55 DEBUG| (monitor qmpmonitor1)    actual: 11752439808
08:13:56 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:56 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'jk0mYZEl'}
08:13:56 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:56 DEBUG| (monitor qmpmonitor1)    actual: 11666456576
08:13:57 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:57 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'LS92XbO3'}
08:13:57 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:57 DEBUG| (monitor qmpmonitor1)    actual: 11561598976
08:13:58 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:58 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'cYnEtvIB'}
08:13:58 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:58 DEBUG| (monitor qmpmonitor1)    actual: 11460935680
08:13:59 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:13:59 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'iEq4lQQn'}
08:13:59 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:13:59 DEBUG| (monitor qmpmonitor1)    actual: 11381243904
08:14:00 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:14:00 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'wgaBzUB5'}
08:14:00 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:14:00 DEBUG| (monitor qmpmonitor1)    actual: 11299454976
08:14:01 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:14:01 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'bfhEevkQ'}
08:14:01 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:14:01 DEBUG| (monitor qmpmonitor1)    actual: 11213471744
08:14:02 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:14:02 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'wruvCYUR'}
08:14:02 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:14:02 DEBUG| (monitor qmpmonitor1)    actual: 11119099904
08:14:03 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
08:14:03 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'j0vgLsSh'}
08:14:03 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
08:14:03 DEBUG| (monitor qmpmonitor1)    actual: 11026825216
08:16:15 ERROR| Reproduced traceback from: /usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/avocado_vt/test.py:420
08:16:15 ERROR| Traceback (most recent call last):
08:16:15 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-45.0-py2.6.egg/avocado_vt/test.py", line 206, in runTest
08:16:15 ERROR|     raise exceptions.TestFail(details)
08:16:15 ERROR| TestFail: Failed to balloon memory to expect value during 100.0s    [context: Change VM memory to 4756]

**Comment by suqinhuang** [comment]
run more than 500 times with win2012r2 and win8.1, could not reproduce it

**Comment by suqinhuang** [comment]
https://github.com/avocado-framework/avocado-vt/pull/892/files

**Comment by suqinhuang** [comment]
@pingl2016 update balloon_timeout and cmd timeout, could you have a try

**Comment by pingl2016** [comment]
met the issue as below:
22:56:24 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-46.0-py2.7.egg/avocado_vt/test.py:420
22:56:24 ERROR| Traceback (most recent call last):
22:56:24 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-46.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
22:56:24 ERROR|     self._runTest()
22:56:24 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-46.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
22:56:24 ERROR|     run_func(self, params, env)
22:56:24 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-46.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
22:56:24 ERROR|     return fn(*args, **kwargs)
22:56:24 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_stress.py", line 92, in run
22:56:24 ERROR|     stop_video(vm)
22:56:24 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_stress.py", line 69, in stop_video
22:56:24 ERROR|     session.cmd(clean_cmd)
22:56:24 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1211, in cmd
22:56:24 ERROR|     raise ShellCmdError(cmd, s, o)
22:56:24 ERROR| ShellCmdError: Shell command failed: 'taskkill /IM "wmplayer.exe" /F'    (status: 128,    output: 'ERROR: The process "wmplayer.exe" not found.\n\n')

**Comment by pingl2016** [comment]
Balloon failed maybe caused by the value set is lower than the used memory of guest.

[balloon_timeout.tar.gz](https://github.com/autotest/tp-qemu/files/826636/balloon_timeout.tar.gz)

File: qemu/tests/balloon_stress.py [hunk 1]
```diff
@@ -34,27 +77,22 @@ def run(test, params, env):
...
-    video_play.start()
-    error_context.context("Run video background", logging.info)
-    check_playing_cmd = params["check_playing_cmd"]
-    running = utils_misc.wait_for(
-        lambda: utils_misc.get_guest_cmd_status_output(
-            vm, check_playing_cmd)[0] == 0, first=60, timeout=600)
-    if not running:
-        raise exceptions.TestError("Video is not playing")
-
+    run_video(vm)
```
Comment (pingl2016) [inline qemu/tests/balloon_stress.py]: If we doing a memory balloon immediately after playing video, QMP event BALLOON_CHANGE may not be received. It is better to sleep a few seconds.

**Comment by suqinhuang** [comment]
PASS 1-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_stress

**Comment by kimi1978wy** [comment]
@suqinhuang 
1 I tested on win8.1-64, it stop video after the 14th time ballooning. and after 5min, it play video again. Could it play video continuously? 
03:56:23 DEBUG| Sending command: del /f /s "%ProgramFiles(x86)%\Windows Media Player\wmplayer.exe"
03:56:23 DEBUG| Sending command: echo %errorlevel%
03:56:42 DEBUG| (monitor qmpmonitor1) Sending command 'query-balloon'
03:56:42 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'ctUn4sk5'}
03:56:42 DEBUG| (monitor qmpmonitor1) Response to 'query-balloon' (re-formated)
03:56:42 DEBUG| (monitor qmpmonitor1)    actual: 16740515840
**03:56:42 INFO | repeat times: 15**
03:56:42 INFO | Context: Change VM memory to 5493
03:56:42 DEBUG| (monitor qmpmonitor1) Sending command 'balloon'
03:56:42 DEBUG| Send command: {'execute': 'balloon', 'arguments': {'value': 5759827968}, 'id': 'STGo29CD'}
03:56:43 INFO | BALLOON_CHANGE QMP event received

**04:01:47 INFO | repeat times: 20**
04:01:50 DEBUG| Sending command: "%ProgramFiles(x86)%\Windows Media Player\wmplayer.exe" "http://fileshare.englab.nay.redhat.com/pub/section2/kvmauto/video/big_buck_bunny_480p_stereo.avi" /play /fullscreen

2 when I tested on win8.1-32. it boot guest with 16G, but we should boot guest with 3G in 32bit os and the balloon value should 0-3G.

**Comment by kimi1978wy** [comment]
@suqinhuang please help change balloon time to 500. 
And sleep time should be removed in this case, since it will cost too much time to check the balloon status.

For this patch , it could run video the first 500 times balloon.

**Comment by suqinhuang** [comment]
Hi wyu:

1. repeat_time is update to 500
2. maxmem issue reproduce, as the design of add_memory in avocado_vt is updated, need to discuss about it
3. sleep_time issue is fixed here https://github.com/autotest/tp-qemu/pull/977/files

**Comment by kimi1978wy** [comment]
@suqinhuang the second time happened this problem, when stop video the first time, it will wait for  5min to start it again. When the balloon is fast, it will miss 100 times balloon evict/enlarge. Please help to check.

**03:49:24** INFO | Context: Stop video
03:49:24 DEBUG| Sending command: del /f /s "%ProgramFiles(x86)%\Windows Media Player\wmplayer.exe"

**03:54:35** DEBUG| Sending command: "%ProgramFiles(x86)%\Windows Media Player\wmplayer.exe" "http://fileshare.englab.nay.redhat.com/pub/section2/kvmauto/video/big_buck_bunny_480p_stereo.avi" /play /fullscreen

**Comment by suqinhuang** [comment]
https://github.com/autotest/tp-qemu/pull/979

**Comment by kimi1978wy** [comment]
after applying 979, it could re-play video in 10s.

Acked-by: Yu Wang wyu@redhat.com

**Comment by suqinhuang** [comment]
@pingl2016 do you have more comment?

File: qemu/tests/balloon_stress.py [hunk 2]
```diff
@@ -34,27 +79,28 @@ def run(test, params, env):
...
+        logging.info("repeat times: %d" % i)
+        play_duration = time.time() - start_time
+        if i == 0 or (play_duration > time_for_video):
+            no_video = utils_misc.wait_for(lambda: not check_video_status(vm),
+                                           30, 5, 2)
+            if not no_video:
+                process = check_video_status(vm)
+                stop_video(process)
+            run_video(vm)
+            start_time = time.time()
```
Comment (luckyh) [inline qemu/tests/balloon_stress.py]: It doesn't make sense to control the video play workflow here, suggest moving into `run_video`.

File: qemu/tests/balloon_stress.py [hunk 3]
```diff
@@ -34,27 +79,28 @@ def run(test, params, env):
 
     driver_name = params["driver_name"]
     utils_test.qemu.setup_win_driver_verifier(driver_name, vm, timeout)
+    run_video(vm)
+    time.sleep(5)
```
Comment (luckyh) [inline qemu/tests/balloon_stress.py]: What is the reason for requesting a sleep here?

**Comment by luckyh** [comment]
@suqinhuang do you have a test result?

File: qemu/tests/balloon_stress.py [hunk 4]
```diff
@@ -18,12 +20,59 @@ def run(test, params, env):
     4) check device using right driver in guest.
     5) play online video in guest
     6) balloon memory in monitor in loop
-    7) check vm alive
+    7) kill video process and run it again (option)
```
Comment (CongLi) [inline qemu/tests/balloon_stress.py]: optional

File: qemu/tests/cfg/balloon_stress.cfg [hunk 5]
```diff
@@ -3,15 +3,18 @@
...
+    balloon = balloon0
+    balloon_dev_devid = balloon0
+    balloon_dev_add_bus = yes
     video_url = http://FILESHARE.COM/pub/section2/kvmauto/video/big_buck_bunny_480p_stereo.avi
     x86_64:
         program_files = "%ProgramFiles(x86)%"
     i386:
         program_files = "%ProgramFiles%"
-    check_playing_cmd = "tasklist|findstr /I mplayer"
+    target_process = "\w+mplayer.exe"
```
Comment (CongLi) [inline qemu/tests/cfg/balloon_stress.cfg]: w+mplayer?

File: qemu/tests/balloon_stress.py [hunk 6]
```diff
@@ -18,12 +20,59 @@ def run(test, params, env):
...
+        check_cmd = params.get("check_process_cmd", "wmic process get name")
+        s, output = utils_misc.get_guest_cmd_status_output(vm, check_cmd)
+        process = re.findall(target_process, output, re.M | re.I)
+        if not process:
+            return None
+        else:
+            logging.info("Video process: %s" % process)
+            return process[0]
+
+    def run_video(vm):
```
Comment (CongLi) [inline qemu/tests/balloon_stress.py]: There is already a sub process when call run_video(), why create a process here again?

File: qemu/tests/cfg/balloon_stress.cfg (hunk 5 above)
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_stress.cfg]: kmplayer or mplayer

**Comment by suqinhuang** [comment]
PASS 1-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_stress

File: qemu/tests/balloon_stress.py [hunk 7]
```diff
@@ -16,14 +16,53 @@ def run(test, params, env):
...
+    5) play video in background repeatly
     6) balloon memory in monitor in loop
-    7) check vm alive
+    7) clean the video
 
     :param test: QEMU test object
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
+    def check_video_status(vm):
```
Comment (luckyh) [inline qemu/tests/balloon_stress.py]: Please move this function into `qemu/tests/win_video_play.py`.

File: qemu/tests/balloon_stress.py [hunk 8]
```diff
@@ -16,14 +16,53 @@ def run(test, params, env):
...
+        """
+        Run video in background
+
+        :param vm: VM object
+        """
+        while True:
+            utils_test.run_virt_sub_test(test, params, env,
+                                         params.get("video_test"))
+
+    def stop_video(vm):
```
Comment (luckyh) [inline qemu/tests/balloon_stress.py]: Shouldn't be removed?

File: qemu/tests/balloon_stress.py (hunk 7 above)
Comment (suqinhuang) [inline qemu/tests/balloon_stress.py]: test won't raise when error happen in thread, so I check process in memory balloon loop

**Comment by suqinhuang** [comment]
PASS 1-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.balloon_stress

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**